wgpu-core/src/device/mod.rs

```diff
@@ -149,14 +149,22 @@ fn map_buffer<B: hal::Backend>(
             }
         }
     }
-
+    buffer.map_state = resource::BufferMapState::Active;
```

Comment:
is this needed here? I thought we agreed to instead *expect* this map state, not assign it

Reply:
This will be need for `device_create_buffer_mapped()`. I'll just move it to that function.